test/test_kmod/hello_mod/hello-1.c

```diff
@@ -0,0 +1,22 @@
+/*
+ *  hello-1.c - The simplest kernel module.
+ */
+#include <linux/module.h>	/* Needed by all modules */
+#include <linux/kernel.h>	/* Needed for KERN_INFO */
+
+int init_module(void)
+{
+	printk(KERN_INFO "Hello world 1.\n");
+
+	/*
+	 * A non 0 return means init_module failed; module can't be loaded.
+	 */
+	return 0;
+}
+
+void cleanup_module(void)
+{
+	printk(KERN_INFO "Goodbye world 1.\n");
+}
+
+MODULE_LICENSE("GPL");
```

Comment:
Why GPL?  The rest of Nix uses MIT.

Reply:
The module is taken from tldp.org where it is licensed GPL.

Reply:
This also isn't specific enough, as the GPL has versions. I assume that means 2.0. Let's get a new one written that we can license MIT. Also, can we name it `hello.c`. That `-1` is really killing me.

Reply:
A pure MIT licensed module would taint the kernel if loaded. The option I see would be: [`Dual MIT/GPL`](https://elixir.bootlin.com/linux/latest/source/include/linux/module.h#L180). I willl create a new module and license it accordingly.